packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -1293,7 +1295,17 @@ export function getBodyType(
       }
     }
     if (resourceType && bodyModel.name === "") {
-      bodyModel = resourceType;
+      const metadataInfo = createMetadataInfo(program);
+      const visibility = resolveRequestVisibility(
+        program,
+        route.operation,
+        route.verb
+      );
+      const payloadType = metadataInfo.getEffectivePayloadType(
```

Comment:
Does this only happen in body type? Is it possible in general property if there is a anonymous model?